app/src/main/java/com/itverse/futuris/activities/ComposantActivity.kt

```diff
@@ -80,32 +106,52 @@ class ComposantActivity : AppCompatActivity() {
         // automatically handle clicks on the Home/Up button, so long
         // as you specify a parent activity in AndroidManifest.xml.
         return when (item.itemId) {
-            R.id.action_save -> getStorageWriteAccess()
+            R.id.action_save -> export()
             else -> super.onOptionsItemSelected(item)
         }
     }
 
-    private fun getStorageWriteAccess(): Boolean {
-        when {
-            ContextCompat.checkSelfPermission(
-                this,
-                android.Manifest.permission.WRITE_EXTERNAL_STORAGE
-            ) == PackageManager.PERMISSION_GRANTED -> {
-                println("Already has permission!")
-                generateExcelFileTest(this, projectSelected)
-                Toast.makeText(this, "Exportation terminé avec success",
-                    Toast.LENGTH_SHORT).show()
-            }
-            shouldShowRequestPermissionRationale(android.Manifest.permission.WRITE_EXTERNAL_STORAGE) -> {
-                println("Running shouldShowRequestPermissionRationale condition")
-            }
-            else -> {
-                // You can directly ask for the permission.
-                // The registered ActivityResultCallback gets the result of this request.
-                requestPermissionLauncher.launch(
-                    android.Manifest.permission.WRITE_EXTERNAL_STORAGE)
-            }
+    private fun export(): Boolean {
+        // when you create document, you need to add Intent.ACTION_CREATE_DOCUMENT
+
+        val intent = Intent(Intent.ACTION_CREATE_DOCUMENT).apply{
+            addCategory(Intent.CATEGORY_OPENABLE)
+            type = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+            putExtra(Intent.EXTRA_TITLE, "Project.xlsx")
```

Comment:
Is "Project.xlsx" the title of the documents? If so can we find a way to use the project's title instead of the hard-coded string?